### ISSUE TITLE
Missing `--softmax-temperature T`

### ISSUE THREAD

**Issue by AmitMY** [awslabs/sockeye#1109]
I would like to sample from my decoder.

The docs say I can:
<img width="997" alt="image" src="https://github.com/awslabs/sockeye/assets/5757359/f409ddb2-8512-4958-bd97-d11accf7b3df">

But I get an error:
```bash
usage: translate.py [-h] [--config CONFIG] [--input INPUT] [--input-factors INPUT_FACTORS [INPUT_FACTORS ...]] [--json-input] [--output OUTPUT] --models MODELS [MODELS ...]
                    [--checkpoints CHECKPOINTS [CHECKPOINTS ...]] [--nbest-size NBEST_SIZE] [--beam-size BEAM_SIZE] [--greedy] [--beam-search-stop {all,first}]
                    [--batch-size BATCH_SIZE] [--chunk-size CHUNK_SIZE] [--sample [SAMPLE]] [--seed SEED] [--ensemble-mode {linear,log_linear}] [--bucket-width BUCKET_WIDTH]
                    [--max-input-length MAX_INPUT_LENGTH] [--max-output-length-num-stds MAX_OUTPUT_LENGTH_NUM_STDS] [--max-output-length MAX_OUTPUT_LENGTH]
                    [--restrict-lexicon RESTRICT_LEXICON [RESTRICT_LEXICON ...]] [--restrict-lexicon-topk RESTRICT_LEXICON_TOPK] [--skip-nvs] [--nvs-thresh NVS_THRESH]
                    [--strip-unknown-words] [--prevent-unk] [--output-type {translation,score,translation_with_score,translation_with_factors,benchmark,json}]
                    [--length-penalty-alpha LENGTH_PENALTY_ALPHA] [--length-penalty-beta LENGTH_PENALTY_BETA] [--brevity-penalty-type {none,learned,constant}]
                    [--brevity-penalty-weight BREVITY_PENALTY_WEIGHT] [--brevity-penalty-constant-length-ratio BREVITY_PENALTY_CONSTANT_LENGTH_RATIO]
                    [--dtype {None,float32,float16,bfloat16,int8}] [--clamp-to-dtype] [--device-id DEVICE_ID] [--use-cpu] [--env ENV] [--tf32 TF32] [--quiet]
                    [--quiet-secondary-workers] [--no-logfile] [--loglevel {INFO,DEBUG,ERROR}] [--loglevel-secondary-workers {INFO,DEBUG,ERROR}] [--knn-index KNN_INDEX]
                    [--knn-lambda KNN_LAMBDA]
translate.py: error: unrecognized arguments: --softmax-temperature 2.0
```

Is this supported otherwise?

**Comment by mjdenkowski** [comment]
Hi Amit,

This was implemented in the old MXNet version of Sockeye. When we updated Sockeye to PyTorch ([3.1.0](https://github.com/awslabs/sockeye/blob/98b49469afb7c47a059529818464ba130e66b65f/CHANGELOG.md#310)), `--softmax-temperature` was one of the features we chose not to migrate. If you're interested in taking a look at the MXNet code for this feature, the pull request for that implementation is #869.

Best,
Michael
